test/decl/protocol/special/comparable/comparable_unsupported_type.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %target-typecheck-verify-swift -verify-ignore-unknown
+
+// Automatic synthesis is only supported for enums for now.
+
+struct NotComparableStruct: Comparable {
+  // expected-error@-1 {{type 'NotComparableStruct' does not conform to protocol 'Comparable'}}
+  // expected-note@-2 {{automatic synthesis of 'Comparable' is not supported for structs}}
+  var value = 0
+}
+
+class NotComparableClass: Comparable {
+  // expected-error@-1 {{type 'NotComparableClass' does not conform to protocol 'Comparable'}}
+  // expected-note@-2 {{automatic synthesis of 'Comparable' is not supported for classes}}
+  // expected-error@-3 {{type 'NotComparableClass' does not conform to protocol 'Equatable'}}
+  // expected-note@-4 {{automatic synthesis of 'Equatable' is not supported for classes}}
+  var value = 1
+}
+
+// Make sure we don't emit the note for enums.
+
+enum NotComparableEnum: Int, Comparable {
```

Comment:
@theblixguy @xedin Do you think its is worth a note mentioning that enum with raw values cannot synthesize comparable? Maybe its something to throw in some educational notes explaining that ... what you think?
The question is more because if we have a diagnostic that says that comparable conformances can only be synthesized for enums and then have an enum that fails synthesize/conformance, it may be confusing(the why is failing) especially for beginners. Do you thinks this makes sense and its worth giving a bit more information in this case? Perhaps the educational notes would be good for something like this :)

Reply:
I think it makes sense, we could emit a different note to tell the user that, something like `raw values prevent automatic 'Comparable' synthesis` or `enum 'Foo' has raw values, preventing synthesized conformance to `Comparable`. WDYT?

Reply:
We could also add another note for payload cases where the payload does not conform to `Comparable`. I believe we already have a diagnostic note for it, just needs to be applied in this case as well.

Reply:
> I think it makes sense, we could emit a different note to tell the user that, something like `raw values prevent automatic 'Comparable' synthesis` or `enum 'Foo' has raw values, preventing synthesized conformance to 'Comparable'`. WDYT?

To me, the former seems good :)  

> We could also add another note for payload cases where the payload does not conform to Comparable. I believe we already have a diagnostic note for it, just needs to be applied in this case as well.

That would be great! Now that you mention this I'm thinking even more that beyond the note this also could benefit of having an educational note explaining the rules around comparable synthesis, not here in this patch, but something to think about adding in the future :)

Reply:
Alright, I have added the notes. Let me know what you think :) And yeah it would be good to add some educational notes in a follow-up!

Reply:
Right! The note looks good to me :)